vowpalwabbit/ezexample.h

```diff
@@ -131,43 +129,44 @@ class ezexample
   {
     if (vw_ref->ignore_some && vw_ref->ignore[static_cast<int>(c)]) return true;
     if (ns_exists[static_cast<int>(c)]) return false;
-    ec->indices.push_back(static_cast<size_t>(c));
+    std::string s(1, c);
+    fid hash = static_cast<fid>(VW::hash_space(*vw_ref, s));
+    ec->feature_space.get_or_create_feature_group(hash, c);
     ns_exists[static_cast<int>(c)] = true;
     return false;
   }
 
   void addns(char c)
   {
     if (ensure_ns_exists(c)) return;
-
-    ec->feature_space[static_cast<int>(c)].clear();
     past_seeds.push_back(current_seed);
     current_ns = c;
     str[0] = c;
     current_seed = static_cast<fid>(VW::hash_space(*vw_ref, str));
+    ec->feature_space[current_seed].clear();
   }
 
   void remns()
   {
-    if (ec->indices.empty())
+    if (ec->feature_space.empty())
     {
       current_seed = 0;
       current_ns = 0;
     }
     else
     {
-      if (ns_exists[static_cast<int>(current_ns)])
+      if (ns_exists[current_seed])
```

Comment:
ns_exists is still the bool array right? If so, then current_seed will quickly over index it

Reply:
Yes just switched that back